app/controllers/chat_controller.rb

```diff
@@ -143,7 +143,26 @@ def edit_message
 
   def update_user_last_read
     set_channel_and_chatable
-    set_user_last_read
+    UserChatChannelMembership
+      .where(user: current_user, chat_channel: @chat_channel)
+      .update_all(last_read_message_id: params[:message_id])
```

Comment:
I know this is 99% of the time supposed to update only 1 row, but there are a couple of addition checks we usually add to avoid anything terrible from happening.

The first one is to ensure we're updating the rows that needs to be updated by ensuring the value we're updating is actually different from the one stored in the database.

The second one is to ensure we're only ever updating 1 row using `LIMIT 1`.

Not sure if we can easily write that in ActiveRecord, but in SQL it should look something like this

```sql
  UPDATE user_chat_channel_membership
     SET last_read_message_id = @params[:message_id]
   WHERE user_id = @current_user.id
     AND chat_channel_id = @chat_channel.id
     AND last_read_message_id <> @params[:message_id]
   LIMIT 1
```

Reply:
This logic has a couple of bugs. I fixed them in #862 and will remove this change from here.